### PR TITLE
build: Add support to build libdbus as fallback

### PR DESCRIPTION
**Pull request by igaw**
The dbus project supports to be consumed via Meson subproject mechanism.  Update the build configuration accordingly.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -89,8 +89,24 @@ if openssl_dep.found()
            description: 'OpenSSL/LibreSSL API version @0@'.format(api_version))
 endif
 
-# Check for libdus availability. Optional, only required for MCTP dbus scan
-libdbus_dep = dependency('dbus-1', required: false)
+if get_option('libdbus').disabled()
+    libdbus_dep = dependency('', required: false)
```
Comment (martin-belanger) [inline meson.build:93]: Could we use `declare_dependency` instead?

Comment (igaw) [reply meson.build:93]: Sounds like a good idea according the docs. We have some more of those. I add a separate commit to change all in one go

Comment (igaw) [reply meson.build:93]: Nope, doesn't work. The `declare_dependecy` dependency object will return true `.found()`.

Comment (martin-belanger) [reply meson.build:93]: Good to know...

Comment (igaw) [reply meson.build:93]: And one thing I don't understand is that the documentation talks about internal build but we have to use declare_dependency to make the dependency being announced to be used in a wrap... I am too stupid too understand all these things :)

Comment (martin-belanger) [reply meson.build:93]: I read the documentation and did not understand as well... :smirk:

**Pull request merged**